sw/device/rom_exts/rom_ext_manifest_parser.h

```diff
@@ -0,0 +1,213 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_ROM_EXTS_ROM_EXT_MANIFEST_PARSER_H_
+#define OPENTITAN_SW_DEVICE_ROM_EXTS_ROM_EXT_MANIFEST_PARSER_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/rom_exts/manifest.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * ROM Extension image signature.
+ */
+typedef struct rom_ext_signature {
+  uint32_t data[ROM_EXT_IMAGE_SIGNATURE_SIZE_WORDS];
+} rom_ext_signature_t;
+
+/**
+ * ROM Extension lockdown information.
+ *
+ * TODO - probably would eventually become an internal type, and there will
+ *        be another public type with more finely parsed information.
+ */
+typedef struct rom_ext_lockdown_info {
+  uint32_t data[ROM_EXT_PERIPHERAL_LOCKDOWN_INFO_SIZE_WORDS];
+} rom_ext_lockdown_info_t;
+
+/**
+ * ROM Extension public key.
+ */
+typedef struct rom_ext_public_key {
+  uint32_t data[ROM_EXT_SIGNATURE_PUBLIC_KEY_SIZE_WORDS];
+} rom_ext_public_key_t;
+
+/**
+ * ROM Extension image extension.
+ */
+typedef struct rom_ext_extension {
+  uint32_t offset;
```

Comment:
This should be a `void*` as it's pointing into the current rom_ext_manifest.

Reply:
Done.